src/MarginTrading.SettingsService.Services/MarketDayOffService.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using MarginTrading.SettingsService.Core;
+using MarginTrading.SettingsService.Core.Domain;
+using MarginTrading.SettingsService.Core.Services;
+using MarginTrading.SettingsService.StorageInterfaces.Repositories;
+using Microsoft.Extensions.Internal;
+
+namespace MarginTrading.SettingsService.Services
+{
+    public class MarketDayOffService : IMarketDayOffService
+    {
+        private readonly IScheduleSettingsRepository _scheduleSettingsRepository;
+        private readonly ISystemClock _systemClock;
+        
+        public MarketDayOffService(
+            IScheduleSettingsRepository scheduleSettingsRepository,
+            ISystemClock systemClock)
+        {
+            _scheduleSettingsRepository = scheduleSettingsRepository;
+            _systemClock = systemClock;
+        }
+
+        public async Task<Dictionary<string, bool>> MarketsStatus(string[] marketIds)
+        {
+            var scheduleSettings = (await _scheduleSettingsRepository.GetFilteredAsync())
+                .Where(x => !string.IsNullOrWhiteSpace(x.MarketId))
+                .Cast<ScheduleSettings>()
+                .GroupBy(x => x.MarketId)
+                .ToDictionary(x => x.Key, x => x.ToList());
+            var currentDateTime = _systemClock.UtcNow.UtcDateTime;
+
+            var rawPlatformSchedule = scheduleSettings.TryGetValue(Constants.PlatformMarketId, out var platformSettings)
+                ? platformSettings
+                : new List<ScheduleSettings>();
+            var platformCompiledSchedule = CompileSchedule(rawPlatformSchedule, currentDateTime);
+
+            var result = marketIds.Except(scheduleSettings.Keys).ToDictionary(
+                marketWithoutSchedule => marketWithoutSchedule,
+                _ => IsOn(platformCompiledSchedule, currentDateTime));
+
+            foreach (var marketToCompile in marketIds.Except(result.Keys))
+            {
+                var compiledSchedule = CompileSchedule(
+                    scheduleSettings[marketToCompile].Concat(
+                        rawPlatformSchedule.WithRank(int.MaxValue)).ToList(), 
+                    currentDateTime);
+                result.Add(marketToCompile, IsOn(compiledSchedule, currentDateTime));
+            }
+
+            return result;
+        }
+        
+        private bool IsOn(IEnumerable<CompiledScheduleTimeInterval> compiledSchedule, DateTime currentDateTime)
+        {
+            var intersecting = compiledSchedule.Where(x => IsBetween(currentDateTime, x.Start, x.End));
+
+            return intersecting.OrderByDescending(x => x.Schedule.Rank)
+                       .Select(x => x.Schedule).FirstOrDefault()?.IsTradeEnabled ?? true;
+        }
+
+        private static bool IsBetween(DateTime currentDateTime, DateTime start, DateTime end)
+        {
+            return start <= currentDateTime && currentDateTime < end;
```

Comment:
`currentDateTime <= end`

Reply:
There is a convention about time intervals in calculations: we include starting point, and exclude ending.

Reply:
okay